pkg/metric/metric.go

```diff
@@ -37,21 +37,68 @@ var (
 	}
 )
 
-// Type represents the type of a metric e.g. a counter. See
-// https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#metric-types.
+// Type represents the type of the metric. See https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#metric-types.
 type Type string
 
-// Gauge defines a OpenMetrics gauge.
-var Gauge Type = "gauge"
+// TypeN represents the type of the metric as an integer.
+type TypeN int
 
-// Info defines an OpenMetrics info.
-var Info Type = "info"
+// String returns the string representation of the metric type.
+func (i TypeN) String() Type {
+	return MetricTypeNMap[i]
```

Comment:
Shouldn't String() return a String() and not a Type?

Reply:
Ah, right.